flow/src/test/scala/org/alephium/flow/core/TxUtilsSpec.scala

```diff
@@ -1250,6 +1250,107 @@ class TxUtilsSpec extends AlephiumSpec {
     sweep().leftValue is "Not enough ALPH for gas fee in sweeping"
   }
 
+  trait MultiGroupTransactions extends FlowFixture with UnsignedTxFixture {
+    implicit override lazy val blockFlow   = isolatedBlockFlow()
+    override val configValues              = Map(("alephium.broker.broker-num", 1))
+    val (fromPrivateKey, fromPublicKey, _) = genesisKeys(0)
+    val lockupScript                       = LockupScript.p2pkh(fromPublicKey)
+    val unlockScript                       = UnlockScript.p2pkh(fromPublicKey)
+
+    val lockupScriptsByGroup = Map(
+      GroupIndex.unsafe(0) -> lockupScript,
+      GroupIndex.unsafe(1) -> Address.p2pkh(GroupIndex.unsafe(1).generateKey._2).lockupScript,
+      GroupIndex.unsafe(2) -> Address.p2pkh(GroupIndex.unsafe(2).generateKey._2).lockupScript
+    )
+
+    val validation = TxValidation.build
+
+    def splitGenesisUtxosBy(n: Int): AVector[AssetOutputInfo] = {
+      val block =
+        transfer(blockFlow, fromPrivateKey, amount = genesisBalance / n, to = fromPublicKey)
+      addAndCheck(blockFlow, block)
+      val utxos = blockFlow
+        .getUTXOs(Address.p2pkh(fromPublicKey).lockupScript, Int.MaxValue, true)
+        .rightValue
+        .asUnsafe[AssetOutputInfo]
+      assume(utxos.length == n)
+      utxos
+    }
```

Comment:
This method is duplicated in both TxUtils and ServerUtils specs and I'm thinking of moving it to higher Trait as it is quite useful in tests to start with multiple utxos instead of just one genesis utxo